source/slang/slang-lower-to-ir.cpp

```diff
@@ -1986,7 +1994,12 @@ static String getNameForNameHint(
 
     StringBuilder sb;
     sb.append(parentName);
-    sb.append(".");
+    if (decl->hasModifier<__exportDirectly>()) {
+        sb.append("::");
+    }
+    else {
+        sb.append(".");
+    }
```

Comment:
I do not believe this code should be here.

As discussed earlier, this logic relates to the (ab)use of the `IRNameHintDecoration` as a way of establishing linkage between C++ and Slang, and once we stop using name hints that way, there is no need for the name hint of a symbol to change in this way.

Secondarily, even if we were to consider adding logic like this for *mangled* names, we'd run into the problem that the use of `::` as a static-lookup-in-scope operation is language-specific (basically just C++ and languages that strive to match C++). Having a bare `::` in the middle of a symbol name only helps when emitting C++ (and even then perhaps only on narrow cases), and for many other potentially-interesting output languages the emit logic would need to parse the symbol name looking for `::`s and replace them with something suitable anyway.

The use of `.` in Slang is intentional as it provides us with a single separator character for all scoped names that is supported and encouraged by the Slang language itself. The `.` also has the advantage of being a single ASCII character, so that it can quickly be scanned for in a symbol name. If the emit logic for a target *did* want to do something like map `.`s to `::`s, they could do it. In fact, when generating HLSL/GLSL-compatible symbol names from a name hint, we already replace `.` with `_`.